gnovm/pkg/gnolang/nodes.go

```diff
@@ -1132,8 +1132,13 @@ func ReadMemPackage(dir string, pkgPath string) *std.MemPackage {
 				Body: string(bz),
 			})
 	}
-	validatePkgName(string(pkgName))
-	memPkg.Name = string(pkgName)
+
+	// If no .gno files are present, package simply does not exist.
+	if len(memPkg.Files) > 0 {
```

Comment:
```suggestion
	if !memPkg.HasGnoFiles() {
```

I think that a new helper would be a good fit here.

Reply:
`Validate()` already exists and does something different, so I vote for `HasGnoFiles()` or `IsEmpty/Empty`.

Reply:
I prefer `Validate()` to be one stop solution. After calling `Validate()` i expects that mem package is totally valid and don't require any other checks?

Maybe we can have Validate() and call all funcs like HasGnoFiles() or IsEmpty/Empty inside?

Reply:
> Validate() already exists and does something different, so I vote for HasGnoFiles() or IsEmpty/Empty.

`IsEmpty` is great !

> I prefer Validate() to be one stop solution ... Maybe we can have Validate() and call all funcs like HasGnoFiles() or IsEmpty/Empty inside?

@harry-hov I'm not sure about this. My initial goal was to **skip** the package name validation if no ".gno" files were present in order to avoid a panic.
or maybe something like this, since `Validate` method also check for package name:
```diff
@@ -1133,10 +1132,12 @@ func ReadMemPackage(dir string, pkgPath string) *std.MemPackage {
 			})
 	}
 
-	// If no .gno files are present, package simply does not exist.
+	// skip package validation if no .gno file are present
 	if !memPkg.IsEmpty() {
-		validatePkgName(string(pkgName))
 		memPkg.Name = string(pkgName)
+		if err := memPkg.Validate(); err != nil {
+			panic("mempkg validate error: " + err.Error())
+		}
 	}
 ```

Reply:
@harry-hov, after some checks, I've managed to only use the `Validate` method inside **ReadMemPackage**. However, I had to modify the pkgpath validation regex in 6ce9a3e:

- To support paths with uppercase letters and starting with number in random realm path generated by the test command, such as `gno.land/r/31lSIbPL` (domain should be case insensitive, so that's ok).
- To support `stdlibs` packages like: "testing", "net/url", and so on.

Please let me know if that's ok for you.